test/Semantics/common.sh

```diff
@@ -10,19 +10,18 @@ function die {
   echo "$(basename $0): $*" >&2
   exit 1
 }
+if [[ $# < 1 ]]; then
```

Comment:
This logic is wrong. This means that only an invocation with no arguments will cause the usage string to be printed. Given that now our interface mandates 3 arguments it would be friendlier to print the usage string if less than 3 arguments are given.